### DISCUSSION TITLE
SQLException Bad Allocation

### DISCUSSION THREAD

**Discussion by david99s** [duckdb/duckdb#2457] [Q&A]
Hey, with certain queries like `Select * from lineitem` on SF=10, i always get a SQLEXception: Bad Allocation. I was really surprised since i thought that DuckDB only fetches small chunks per time to exactly avoid any issues regarding Memoryleaks and so on. 
Do you have any idea how to solve that? Is there something like a setfetchsize like for Postgres?

Thank you very much in advance!

**Comment by Mytherin** [comment]
In principle DuckDB supports lazily fetching chunks, but depending on which API you are using and how you are using the API you might be forcing DuckDB to materialize the entire result. Could you give a more concrete example of what exactly you are doing?

**Comment by david99s** [comment]
I am using the Java JDBC API. In the code below, I just want to measure the time it takes to execute the entire ResultSet of given SQl Statements in .txt-file. In this particular case, the bad allocation error occurs when executing "select * from lineitem"
```
conn = DriverManager.getConnection("jdbc:duckdb:/Users/david/Documents/Studium/Bachelorarbeit/connPostgreSQL/tpch2.db");
         try {BufferedReader br = new BufferedReader(new FileReader("anfragen.txt"));
        String sline="";
        while((sline=br.readLine()) != null){
             try {   
                long startTime2 = System.currentTimeMillis();
                Statement b = conn.createStatement();
                ResultSet rs = b.executeQuery(sline);
                while(rs.next()){};
                rs.close();
                long endtime2 = System.currentTimeMillis();
                System.out.println(counter+". Anfrage: DuckDb " + (endtime2- startTime2) +" ms");
                b.close();
            } catch (OutOfMemoryError e) {
                System.err.println("Java Heap Space Error");
                System.err.println("Maximum JVM memory: " + Runtime.getRuntime().freeMemory());
            }
    conn.close();

    } catch (FileNotFoundException f){
        f.printStackTrace();
    } catch (IOException i){
        i.printStackTrace();
    }
```

**Comment by hannes** [comment]
Should be fixed in https://github.com/duckdb/duckdb/pull/2459

**Reply by david99s** [reply]
Thank you ! But sadly the error still occurs. Here i provide you the exact error response from the terminal
```
Exception in thread "main" java.sql.SQLException: bad allocation
        at org.duckdb.DuckDBNative.duckdb_jdbc_execute(Native Method)
        at org.duckdb.DuckDBPreparedStatement.execute(DuckDBPreparedStatement.java:89)
        at org.duckdb.DuckDBPreparedStatement.executeQuery(DuckDBPreparedStatement.java:100)
        at org.duckdb.DuckDBPreparedStatement.executeQuery(DuckDBPreparedStatement.java:128)
        at test.main(test.java:55)
```

**Reply by hannes** [reply]
Are you sure you tried with my branch?

**Reply by david99s** [reply]
No actually i didnt. I don't really know how to get it into my program? Do i have to download DuckDB again to get the new update?
